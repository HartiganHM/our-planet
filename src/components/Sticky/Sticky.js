import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import './Sticky.css';

class Sticky extends Component {
  componentDidMount() {
    const setInitialHeights = elements => {
      [].forEach.call(elements, (sticky) => {
        sticky.setAttribute('data-sticky-initial', sticky.getBoundingClientRect().top);
      });
    };

    const stickies = document.querySelectorAll('[data-sticky]');
    setInitialHeights(stickies);

    document.addEventListener('scroll', () => {
      const top = document.documentElement.scrollTop || document.body.scrollTop;
      const bottom = document.documentElement.scrollHeight || document.body.scrollHeight;
    })

    [].forEach.call(stickies, (sticky) => {
      const stickyInitial = parseInt(sticky.getAttribute('data-sticky-initial'), 10);
      const stickyEnter = parseInt(sticky.getAttribute('data-sticky-enter'), 10) || stickyInitial;
      const stickyExit = parseInt(sticky.getAttribute('data-sticky-exit'), 10) || bottom;

      if (top >= stickyEnter && top <= stickyExit) {
        sticky.classList.add('sticky');
      } else {
        sticky.classList.remove('sticky');
      }
    })
  }

  render() {
    const { className, enter, exit, children } = this.props;
    return (
      <div
        className={`Sticky ${className}`}
        data-sticky
        data-sticky-enter={enter}
        data-stick-exit={exit}
      >
        {children}
      </div>
    );
  }
}

Sticky.propTypes = {
  className: PropTypes.string,
  enter: PropTypes.string,
  exit: PropTypes.string,
  children: PropTypes.node
};
