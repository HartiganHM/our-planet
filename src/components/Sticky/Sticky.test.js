import React from 'react';
import { shallow } from 'enzyme';
import Sticky from './Sticky';

describe('Sticky tests', () => {
  it('Should match the snapshot', () => {
    const mockLocation = { location: { pathname: '/' } };
    const renderedSticky = shallow(<Sticky location={mockLocation} />);

    expect(renderedSticky).toMatchSnapshot();
  });
});
