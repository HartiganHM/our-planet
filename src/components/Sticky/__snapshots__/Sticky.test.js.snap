// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticky tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sticky
    location={
        Object {
            "pathname": "/",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Header)
        location="/"
/>,
      "className": "Sticky undefined",
      "data-stick-exit": undefined,
      "data-sticky": true,
      "data-sticky-enter": 768,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "location": "/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Header)
          location="/"
/>,
        "className": "Sticky undefined",
        "data-stick-exit": undefined,
        "data-sticky": true,
        "data-sticky-enter": 768,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
